src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,21 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && !isset($file->data[$key]) && $member_name !== null) {
```

Comment:
Here check for `!isset($file->data[$key])` but inside in another IF you check `isset($file)`. If no `$file` variable is defined, then this trait business shouldn't be happening at all if I'm not mistaken.